### DISCUSSION TITLE
Does sglang Support Inference for Multiple Models at the Compilation Stage?

### DISCUSSION THREAD

**Discussion by DopeorNope-Lee** [sgl-project/sglang#1263] [Q&A] [answered]
Hello,

I am writing to ask the following question.

Rather than parallelizing kernels or servers to aggregate the results of multiple models, I would like to know if sglang supports ensembling the results of multiple models at the compilation stage through multiprocessing or other methods.

Specifically, I am asking if it is possible to parallelize and use multiple quantized models (in GGUF, GPTQ, AWQ formats) within a single code using multiprocessing.

Thank you

**Comment by merrymercy** [comment, marked as answer]
I think it is possible. You can use `-mem-fraction-static` to control the memory usage of a server and use multiprocessing/popen to launch many servers https://github.com/sgl-project/sglang/blob/13f1357ef000e0d9dcf6f13dd178d809126f3ac7/test/srt/test_eval_accuracy_mini.py#L19
